app/ante/fee_checker.go

```diff
@@ -11,41 +15,28 @@ const (
 	priorityScalingFactor = 1_000_000
 )
 
-// checkTxFeeWithValidatorMinGasPrices implements the default fee logic, where the minimum price per
-// unit of gas is fixed and set by each validator, and the tx priority is computed from the gas price.
-func checkTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+// CheckTxFeeWithGlobalMinGasPrices implements the default fee logic, where the minimum price per
+// unit of gas is fixed and set globally, and the tx priority is computed from the gas price.
+func CheckTxFeeWithGlobalMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
 	feeTx, ok := tx.(sdk.FeeTx)
 	if !ok {
 		return nil, 0, errors.Wrap(sdkerror.ErrTxDecode, "Tx must be a FeeTx")
 	}
 
-	feeCoins := feeTx.GetFee()
+	fee := feeTx.GetFee().AmountOf(appconsts.BondDenom)
 	gas := feeTx.GetGas()
 
-	// Ensure that the provided fees meet a minimum threshold for the validator,
-	// if this is a CheckTx. This is only for local mempool purposes, and thus
-	// is only ran on check tx.
-	if ctx.IsCheckTx() {
-		minGasPrices := ctx.MinGasPrices()
-		if !minGasPrices.IsZero() {
-			requiredFees := make(sdk.Coins, len(minGasPrices))
-
-			// Determine the required fees by multiplying each required minimum gas
-			// price by the gas limit, where fee = ceil(minGasPrice * gasLimit).
-			glDec := sdk.NewDec(int64(gas))
-			for i, gp := range minGasPrices {
-				fee := gp.Amount.Mul(glDec)
-				requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
-			}
+	minFee := sdk.NewInt(int64(math.Ceil(appconsts.GlobalMinGasPrice * float64(gas))))
```

Comment:
I'll try do some testing around this to sanity check

Reply:
Yeah that's a good point, I think we should be adding overflow checks

Reply:
Are there some linters / static analysis tools out there that would do it for us?

Reply:
if we use max uint64 as a gas limit and global min fee becomes 1.1 there will be an overflow when we convert it to int64 since uint64 does not need to reserve any bits for negative numbers max uint64 is inherintly larger. @cmwaters @evan-forbes

I think this could be resolved by incorporating bigInt. we could also throw an error if a gas limit is larger than the max int64.

Reply:
I'm not sure if big.Int would work unless you scale up the `MinimumGasPrice` to an int. Maybe copy the same `priorityScalingFactor` which is `1_000_000`.

Reply:
We will need to check that gas <= math.MaxInt64

Reply:
but even then if gas is `math.MaxInt64-1` and globalMinGasPrice is > 1 we could still overflow.

Reply:
we could check that `minFee<math.MaxInt64-1`

Reply:
I had something like this in mind
```
if gas > math.MaxInt64 {
    return err
}
minFee := sdk.NewInt(int64(globalMinGasPrice * priorityScalingFactor).MulRaw(int64(gas)).QuoRaw(priorityScalingFactor)
```

Reply:
Check if the division rounds down. If so, we will need to change GTE to GT

Reply:
```go
if gas > math.MaxInt64 {
    return err
}
```
I don't think it matters too too much that this a new block validity rule, however we should document that there is a new effective max on gas (and maybe a new if statement)

the most important thing though is that we are [risking non-determinism with a max int64 and a float64](https://stackoverflow.com/questions/74164159/how-to-keep-precision-for-big-numbers-in-golang-when-converting-from-float-to-bi#:~:text=All%20positive%20integers%20up%20to,is%20happening%20in%20your%20case.) 

I'm a `big.Fan` of just using `big.Int` in one way or another for everything, and we probably need to do that here if we want a bullet proof solution

Reply:
> I'm a `big.Fan` of just using `big.Int`

looool I hope this was intentional